### PR TITLE
fix(perf): Use MutableSearch for queries on the landing page

### PR DESCRIPTION
**Pull request by 0Calories**
Fixes a bug with the Performance landing page's transaction search bar, where it would fail to fetch the results of transactions that have a space in them.

This was happening because when searching for a transaction with a space, Discover is not able to properly parse the query string.

### Before
image

### After
image

### CODE REVIEW

File: static/app/components/performance/searchBar.tsx [hunk 1]
```diff
@@ -193,6 +193,12 @@ function SearchBar(props: SearchBarProps) {
   const handleSearch = (query: string, asRawText: boolean) => {
     setSearchResults([]);
     setSearchString(query);
+
+    // Transactions with a space must be wrapped in quotes
+    if (query.includes(' ')) {
+      query = `"${query}"`;
+    }
```
Comment (Zylphrex) [inline static/app/components/performance/searchBar.tsx:200]: There are other cases where this won't handle it correctly. For example, if the transaction name contains quotes. You may want to look into the `MutableSearch` class.

**Review by udameli — COMMENTED** [review]
could you also check if this works when there is a space and a wildcard?

**Comment by 0Calories** [comment]
@udameli It doesn't :(

I'll look into refining this

**Comment by udameli** [comment]
> @udameli It doesn't :(
> 
> I'll look into refining this

maybe https://github.com/getsentry/sentry/blob/master/static/app/views/performance/data.tsx#L395 is a good place to start since this is how we read the transaction filter

**Comment by udameli** [comment]
> > @udameli It doesn't :(
> > I'll look into refining this
> 
> maybe https://github.com/getsentry/sentry/blob/master/static/app/views/performance/data.tsx#L395 is a good place to start since this is how we read the transaction filter

also FWIW in the example that Tony provided, `PUT /profile` filter worked as intended but `GET /organizations/:organization_id/projects/:project_id/flamegraph` didn't so maybe look into why these two filters are handled differently?

**Comment by 0Calories** [comment]
@udameli I was looking into that earlier, I don't think the`POST /profile` filter was working entirely correctly because it's fetching a different transaction as well. I assumed it was just some wackiness with how Discover was parsing the query string

image

**Comment by Zylphrex** [comment]
> > > @udameli It doesn't :(
> > > I'll look into refining this
> > 
> > 
> > maybe https://github.com/getsentry/sentry/blob/master/static/app/views/performance/data.tsx#L395 is a good place to start since this is how we read the transaction filter
> 
> also FWIW in the example that Tony provided, `PUT /profile` filter worked as intended but `GET /organizations/:organization_id/projects/:project_id/flamegraph` didn't so maybe look into why these two filters are handled differently?

Without looking any closer into the code, I'm almost certain the `:` in the query is throwing off the parser somewhere and causing it to interpret the second half of the query with `/organizations/` as the key and `organization_id/projects/:project_id/flamegraph` as the value somehow

**Comment by 0Calories** [comment]
Wrapping the query with `MutableSearch` and calling `formatString()` on it seems to cover all of the above edge cases. Please let me know if there's anything I may have missed 😄

File: static/app/components/performance/searchBar.tsx [hunk 2]
```diff
@@ -193,7 +193,9 @@ function SearchBar(props: SearchBarProps) {
   const handleSearch = (query: string, asRawText: boolean) => {
     setSearchResults([]);
     setSearchString(query);
-    const fullQuery = asRawText ? query : `transaction:${query}`;
+    query = new MutableSearch(query).formatString();
```
Comment (udameli) [inline static/app/components/performance/searchBar.tsx:196]: you might be missing a bracket here
```suggestion
    query = (new MutableSearch(query)).formatString();
```

Comment (0Calories) [reply static/app/components/performance/searchBar.tsx:196]: The formatter removes the brackets if I do that, pretty sure it's still valid syntax as is!

Comment (udameli) [reply static/app/components/performance/searchBar.tsx:196]: really? oh okay then haha

**Review by udameli — APPROVED** [review]

**Pull request merged**